eng/testing/tests.browser.targets

```diff
@@ -35,6 +32,11 @@
     <_BundleAOTTestWasmAppForHelixDependsOn>$(_BundleAOTTestWasmAppForHelixDependsOn);PrepareForWasmBuildApp;_PrepareForAOTOnHelix</_BundleAOTTestWasmAppForHelixDependsOn>
   </PropertyGroup>
 
+  <!-- trimming tests InvariantGlobalizationFalse require full ICU data -->
+  <PropertyGroup Condition="'$(TestTrimming)' == 'true'">
```

Comment:
Why only when 'TestTrimming' ?

Reply:
That's the story:
I made this PR to reduce the number of tests that run with this flag set to true to minimum. However, I was not able to make `InvariantGlobalizationFalse.cs` in trimming tests pass - even setting the Property straight in that test https://github.com/dotnet/runtime/pull/82619/commits/d1f019b826ce3b97480083953c37c655d443f725 was not woring - when we were entering `WasmApp.targets` the value was already trimmed.
I came up with the workaround: I won't set the flag for all the tests in `tests.browser.targets` but only for the trimming ones, to make `InvariantGlobalizationFalse.cs` pass.

Reply:
You also need `'$(IsTrimmingTestProject)' == 'true'`. `$(TestTrimming)` would be set for the whole build, but it is relevant only for trimming projects which will have `$(IsTrimmingTestProject)` set. At least based on reading the code:)

Reply:
Changing it to `IsTrimmingTestProject` breaks the test again. It has to be connected with the order of `*.targets` evaluation, because for the discussed test both flags: `IsTrimmingTestProject` and `TestTrimming` are `true` in the binlog. So it must be that we are reading `test.browser.targets` before we read `Directory.Build.props` where `IsTrimmingTestProject` value is set. `TestTrimming` is set in `src/libraries/tests.proj` and is already availabe in the line quoted.
Conclusion: the condition has to stay as it is.

Reply:
I'm confused about it. Do all combinations 4 of trimming and invariant work for native rebuild ? 
Is this just issue with unit testing ?

Reply:
We took it offline with Pavel and it's not about trimming but about these files being used in different builds. That is why values set in one of them are not reflected in the other.

Reply:
There is nested MSBuild which didn't receive `WasmIncludeFullIcuData` from `System.Runtime.TrimmingTests.proj`

![image](https://user-images.githubusercontent.com/271576/221934519-d5ca18f6-773e-4198-8c9c-fa19d92818fd.png)

And then full file is not there!

![image](https://user-images.githubusercontent.com/271576/221935211-f0d5eb84-e36c-4558-9562-60de4426a1b7.png)